src/hooks/useApprove.ts

```diff
@@ -67,14 +67,8 @@ export const useLotteryApprove = () => {
 export const useIfoApprove = (tokenContract: Contract, spenderAddress: string) => {
   const { account } = useWallet()
   const onApprove = useCallback(async () => {
-    try {
-      const tx = await tokenContract.methods
-        .approve(spenderAddress, ethers.constants.MaxUint256)
-        .send({ from: account })
-      return tx
-    } catch {
-      return false
-    }
+    const tx = await tokenContract.methods.approve(spenderAddress, ethers.constants.MaxUint256).send({ from: account })
+    return tx
```

Comment:
Removed try/catch so the error would not be swallowed

Reply:
The problem was the lack of ui feedback, imo we should keep the try catch and display a notification saying the approval has failed

Reply:
This solves that exact problem. It was swallowing the error not allowing the UI to catch it.

Reply:
Perhaps we should re-throw the error instead.